airbyte-integrations/connectors/destination-csv/src/main/java/io/airbyte/integrations/destination/csv/CsvDestination.java

```diff
@@ -183,14 +189,41 @@ protected void close(boolean hasFailed) throws IOException {
       // do not persist the data, if there are any failures.
       if (!hasFailed) {
         for (final WriteConfig writeConfig : writeConfigs.values()) {
-          Files.move(writeConfig.getTmpPath(), writeConfig.getFinalPath(), StandardCopyOption.REPLACE_EXISTING);
+          final boolean fileAlreadyExists = writeConfig.getFinalPath().toFile().exists();
+          if (writeConfig.getSyncMode() == SyncMode.FULL_REFRESH || !fileAlreadyExists) {
+            Files.move(writeConfig.getTmpPath(), writeConfig.getFinalPath(), StandardCopyOption.REPLACE_EXISTING);
+          } else if (writeConfig.getSyncMode() == SyncMode.INCREMENTAL) {
+            insertCsvFile(writeConfig.getTmpPath(), writeConfig.getFinalPath());
+          }
         }
       }
       // clean up tmp files.
       for (final WriteConfig writeConfig : writeConfigs.values()) {
         Files.deleteIfExists(writeConfig.getTmpPath());
       }
+    }
 
+    /**
+     * Copy and append Csv file to another
+     *
+     * @param srcFilePath CSV file to append data from
+     * @param dstFilePath CSV file to append data to
+     * @throws IOException
+     */
+    private static void insertCsvFile(Path srcFilePath, Path dstFilePath) throws IOException {
```

Comment:
this may be overkill but one thing that would be more robust would be to do the following:
```
* copy existing file to a tmp file
* append new data to the tmp file
* perform a move operation of the tmp file to main file (overwriting the main file).
```

because moves are atomic (on most) filesystems, it guarantees we don't ever end up in a state where we have a failure while writing to the main file and end up writing half a line or something corrupt. i think the current approach is susceptible to this sort of corruption. the new approach would be as close as we could get to being "transactional". wdyt?

Reply:
Great comment, this makes this PR even shorter!!